processing/src/main/java/org/apache/druid/math/expr/ExprMacroTable.java

```diff
@@ -172,7 +172,7 @@ public boolean equals(Object o)
       if (o == null || getClass() != o.getClass()) {
         return false;
       }
-      BaseScalarMacroFunctionExpr that = (BaseScalarMacroFunctionExpr) o;
+      BaseMacroFunctionExpr that = (BaseMacroFunctionExpr) o;
```

Comment:
how important the `equals` / `hashCode` for these classes?
it seems like `ContainsExpr#searchFunction` might not alter the `hashCode`

Reply:
I don't recall how important they are; I noticed this particular bug when writing a test case in another patch. However we should still fix bugs in the methods when they exist. Not altering `hashCode` is fine for correctness I guess but may be a performance issue?